client/state/posts/actions.js

```diff
@@ -68,3 +71,36 @@ export function requestSitePosts( siteId, query = {} ) {
 		} );
 	};
 }
+
+/**
+ * Triggers a network request to fetch a specific post from a site.
+ *
+ * @param  {Number}   siteId Site ID
+ * @param  {Number}   postId Post ID
+ * @return {Function}        Action thunk
+ */
+export function requestSitePost( siteId, postId ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: POST_REQUEST,
+			siteId,
+			postId
+		} );
+
+		return wpcom.site( siteId ).post( postId ).get().then( ( post ) => {
+			dispatch( receivePost( post ) );
+			dispatch( {
+				type: POST_REQUEST_SUCCESS,
+				siteId,
+				postId
+			} );
+		} ).catch( ( error ) => {
```

Comment:
>There's a real subtle diff between the two cases.

Subtle indeed, though still unsure which makes the most sense. Would we want to allow the `yay` case to throw uncaptured, or is it preferable to have the catch block capture errors (dispatching failure action) in both the request and `yay` handler? Obviously we don't want errors in our handlers in any situation, so perhaps not important to dwell too much over.